### PR TITLE
WIP: Add colorizer option to lesspipe

### PR DESCRIPTION
**Pull request by wedens**
See "OUTPUT COLORIZATION" section in lesspipe doc: https://www.gsp.com/cgi-bin/man.cgi?section=1&topic=lesspipe

### CODE REVIEW

File: modules/programs/lesspipe.nix [hunk 1]
```diff
@@ -8,12 +8,25 @@ with lib;
   options = {
     programs.lesspipe = {
       enable = mkEnableOption "lesspipe preprocessor for less";
+
+      colorizer = mkOption {
+        type = types.str;
```
Comment (rycee) [inline modules/programs/lesspipe.nix]: Need `types.nullOr types.str` here otherwise `colorizer = null` is not a valid value.

**Comment by rycee** [comment]
Thanks for the contribution! Added a minor comment. Also please change the commit message to "lesspipe: add colorizer option".

**Comment by wedens** [comment]
I've made the changes, but I don't think this option actually works...

lesspipe seems to do some checks that colorizer is some known value and it doesn't seem to : https://github.com/wofr06/lesspipe/blob/lesspipe/code2color#L3249. 

I've checked on Arch and it just silently ignores colorizer if it has unknown value.

**Comment by rycee** [comment]
Yeah, perhaps the type would have to be an enum instead? Could have an attribute mapping the enum value to the binary:
```nix
colorizers = {
  pygmentize = "${pkgs.python3Packages.pygments}/bin/pygmentize";
  # …
}
```
and the type becomes something like
```nix
types.nullOf types.enum (attrNames colorizers)
```

**Pull request closed without merging**